sys/include/net/gnrc/ipv6.h

```diff
@@ -11,10 +11,63 @@
  * @ingroup     net_gnrc
  * @brief       GNRC's IPv6 implementation
  *
- * The IPv6 control thread understands messages of type
- *
- *  * @ref GNRC_NETAPI_MSG_TYPE_RCV, and
- *  * @ref GNRC_NETAPI_MSG_TYPE_SND,
+ * This module is for usage with the @ref net_gnrc_netapi
+ *
+ * # NETAPI command documentation
+ *
+ * ## `GNRC_NETAPI_MSG_TYPE_RCV`
+ *
+ * @ref GNRC_NETAPI_MSG_TYPE_RCV expects a @ref net_gnrc_pkt (referred to as
+ * "packet" in the following) in receive order (payload first, headers in
+ * decreasing order by layer). It should at least contain a link-layer payload.
+ * If it contains a @ref net_gnrc_netif_hdr it will be taken into account.
+ *
+ * If the link-layer payload is not an IPv6 packet (starting with a 0x6
+ * half-byte) or the module is not able to parse it correctly it will be
+ * dropped. If the packet can't be forwarded (either to another node or an
+ * application) it will be dropped. Otherwise, there are two scenarios of
+ * success:
+ *
+ * 1. If the destination address is an address on this host, and a suitable
+ *    application is registered to the next header field of the IPv6 header,
```

Comment:
What is a "suitable application"? Who checks its appropriateness?

Reply:
The application makes itself suitable by registering itself. A person developing for GNRC or reading this sentence should know what this mean.

Reply:
But here as well: I will replace application with "upper layer". I guess I was in an NDN mindset, when I was writing this :sweat_smile:

Reply:
Did that with the rework of the sentence requested in https://github.com/RIOT-OS/RIOT/pull/10583#discussion_r243770438